core/src/main/java/org/apache/accumulo/core/util/SimpleThreadPool.java

```diff
@@ -28,15 +28,16 @@
  */
 public class SimpleThreadPool extends ThreadPoolExecutor {
 
-  public SimpleThreadPool(int max, final String name) {
-    super(max, max, 4L, TimeUnit.SECONDS, new LinkedBlockingQueue<>(),
+  public SimpleThreadPool(int coreAndMax, boolean allowCoreThreadTimeOut, final String name) {
+    super(coreAndMax, coreAndMax, 4L, TimeUnit.SECONDS, new LinkedBlockingQueue<>(),
         new NamingThreadFactory(name));
-    allowCoreThreadTimeOut(true);
+    allowCoreThreadTimeOut(allowCoreThreadTimeOut);
   }
 
-  public SimpleThreadPool(int max, final String name, BlockingQueue<Runnable> queue) {
-    super(max, max, 4L, TimeUnit.SECONDS, queue, new NamingThreadFactory(name));
-    allowCoreThreadTimeOut(true);
+  public SimpleThreadPool(int coreAndMax, boolean allowCoreThreadTimeOut, final String name,
+      BlockingQueue<Runnable> queue) {
+    super(coreAndMax, coreAndMax, 4L, TimeUnit.SECONDS, queue, new NamingThreadFactory(name));
```

Comment:
While we examining this code, I think it would be a good idea to increase this timeout to a few minutes and use a a private constant.   Maybe 3 to 5 mins.  Could do `private static final   TIMEOUT_SECS=180`

```suggestion
    super(coreAndMax, coreAndMax, TIMEOUT_SECS, TimeUnit.SECONDS, queue, new NamingThreadFactory(name));
```

Reply:
I changed this to TIMEOUT_SECS with a value of 180.  I am wondering whether this should be configurable as well.  @keith-turner thoughts?

Reply:
I am not sure it needs to be made configurable.  If it were configurable maybe we could have a single property with just the timeout and a timeout of zero means no timeout.

Reply:
Your idea is starting to grow on me.  Instead of ALLOW_TIMEOUT properties I will create a THREAD_TIMEOUT properties which default to 0.  I will work that through and update unless I find any issues.